fdbserver/VersionedBTree.actor.cpp

```diff
@@ -2654,37 +2723,50 @@ class DWALPager final : public IPager2 {
 		return Void();
 	}
 
+	// The caller must keep page in scope until future is ready
```

Comment:
This function is not an actor so when it returns the reference it holds is dropped.

Reply:
I see. I still confused about Flow from time to time.

Reply:
That said, there actually could be a bug here because the page cache is the holder of the page until the write future is ready, but a double write of the same page (not currently done but in the future it could be, there is a use case for this) might break this because the cache would only hold the latest.  

I think to be sure I will add the page reference to the underlying actor, write PhysicalBlock.